### ISSUE TITLE
ember serve: ENOENT, no such file or directory

### ISSUE THREAD

**Issue by seandeburca** [ember-cli/ember-cli#4661]
Hi, I'm having trouble building my app or running the dev server. Upgrading ember-cli is next on the bucket list. Any help appreciated.

Mac OSX      10.10.3
ember-cli          0.2.7
npm                  2.7.4
node               0.12.1
bower               1.4.1

`$ rm -rf  node_modules/  bower_components/ tmp/ dist/`
`$ npm cache clear`
`$ bower cache clean`
 `$ npm install && bower install && ember serve`

Error: ENOENT, no such file or directory '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-YOlUTrXq.tmp/0/*.js'

Debug Log: 

     broccoli-caching-writer  cache eviction due to:
     - {directory, ., 16832, 170, 170, 1439394968000}
     - {undefined} +0ms
       broccoli-sane-watcher addWatchDir: /workspace/client/node_modules/ember-intl/node_modules/intl +16s
      broccoli-funnel:Funnel processFilters { in: '17ms', filesFound: 3, filesProcessed: 3, inputPath: '/workspace/client/tmp/funnel-input_base_path-fydXsPOU.tmp/0/', destPath: '/Users/seandeburca/workspace/client/tmp/funnel-output_path-p7uMSd7Y.tmp/assets/intl/polyfill' } +27s
      broccoli-funnel:Funnel build, { in: '17ms', linkedRoots: false, inputPath: '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-fydXsPOU.tmp/0/', destPath: '/Users/seandeburca/workspace/client/tmp/funnel-output_path-p7uMSd7Y.tmp/assets/intl/polyfill' } +0ms
      broccoli-sane-watcher addWatchDir: /Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/intl/locale-data/jsonp +23ms
      broccoli-sane-watcher triggerError { [Error: ENOENT, no such file or directory '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-YOlUTrXq.tmp/0/*.js'] errno: -2, code: 'ENOENT', path: '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-YOlUTrXq.tmp/0/*.js', syscall: 'lstat' } +49ms
      ember-cli:watcher didError { [Error: ENOENT, no such file or directory '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-YOlUTrXq.tmp/0/*.js'] errno: -2, code: 'ENOENT', path: '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-YOlUTrXq.tmp/0/*.js', syscall: 'lstat' } +1ms
    ENOENT, no such file or directory '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-YOlUTrXq.tmp/0/*.js'
    Error: ENOENT, no such file or directory '/Users/seandeburca/workspace/client/tmp/funnel-input_base_path-YOlUTrXq.tmp/0/*.js'
    at Error (native)
    at Object.fs.lstatSync (fs.js:796:18)
    at symlink (/Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/node_modules/symlink-or-copy/index.js:63:26)
    at Function.symlinkOrCopySync     (/Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/node_modules/symlink-or-copy/index.js:58:5)
    at Funnel._copy (/Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/index.js:252:17)
    at Funnel.processFile (/Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/index.js:243:8)
    at Funnel.processFilters (/Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/index.js:154:12)
    at Funnel.build (/Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/index.js:118:10)
    at /Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/node_modules/broccoli-plugin/read_compat.js:61:34
    at lib$rsvp$$internal$$tryCatch (/Users/seandeburca/workspace/client/node_modules/ember-intl/node_modules/broccoli-funnel/node_modules/broccoli-plugin/node_modules/promise-map-series/node_modules/rsvp/dist/rsvp.js:493:16)`

package.json:

       {
        "name": "anonymous",
        "version": "anonymous",
        "description": "anonymous",
        "private": true,
        "directories": {
        "doc": "doc",
        "test": "tests"
      },
      "scripts": {
        "start": "ember server",
        "build": "ember build",
        "test": "ember test"
      },
      "repository": "",
      "engines": {
        "node": ">= 0.10.0"
      },
      "author": "",
      "license": "MIT",
      "devDependencies": {
        "ember-sprite": "0.0.4",
        "body-parser": "^1.12.2",
        "broccoli-asset-rev": "^2.0.0",
        "broccoli-sass": "^0.6.0",
        "connect-restreamer": "^1.0.2",
        "ember-cli": "~0.2.1",
        "ember-cli-app-version": "~0.3.3",
        "ember-cli-babel": "^4.0.0",
        "ember-cli-bootstrap-breakpoints": "1.0.0",
        "ember-cli-build-notifications": "0.0.1",
        "ember-cli-content-security-policy": "0.4.0",
        "ember-cli-dependency-checker": "0.0.8",
        "ember-cli-htmlbars": "0.7.4",
        "ember-cli-i18n": "0.0.6",
        "ember-cli-ic-ajax": "0.1.1",
        "ember-cli-inject-live-reload": "^1.3.0",
        "ember-cli-inline-content": "~0.4.0",
        "ember-cli-make-buildinfo": "git://github.com/mackers/ember-cli-make-buildinfo.git",
        "ember-cli-qunit": "0.3.9",
        "ember-cli-uglify": "1.0.1",
        "ember-data": "1.0.0-beta.16",
        "ember-devtools": "^3.0.4",
        "ember-export-application-global": "^1.0.2",
        "ember-inflector": "~1.5.1",
        "ember-intl": "^1.2.2",
        "ember-validations": "2.0.0-alpha.3",
        "express": "^4.13.3",
        "glob": "^4.5.3",
        "http-proxy": "^1.8.1",
        "morgan": "^1.6.1",
        "rimraf": "~2.2.8"
      }
    }

**Comment by stefanpenner** [comment]
> TrXq.tmp/0/*.js'

Indicates some glob is leaking through, please provide an isolated repo so I can quickly reproduce and explore a fix.

**Comment by demersus** [comment]
Here is a bare app with a couple dependencies added. https://github.com/demersus/ember-funnel-errors
I am experiencing a very similar error message.

**Comment by stefanpenner** [comment]
> https://github.com/demersus/ember-funnel-errors
> I am experiencing a very similar error message.

thanks, looking now.

**Comment by stefanpenner** [comment]
This appears to be a bug in the version of ember-intl you are using.

specifically: https://github.com/yahoo/ember-intl/blob/1.3.0/index.js#L91
the bug appears to be fixed (in the 2.0 betas) https://github.com/yahoo/ember-intl/blob/master/index.js#L164

This issue has manifested itself likely do to recent broccoli-funnel changes, which exposed the bug.

relevant Funnel docs:

```
files {Array of Strings}
One or more relative file paths. Files within the node whose relative paths match will be copied (with the location inside their parent directories preserved) to the destDir.
Default: [].
```


In the 1.3.0, Funnel is using `files` incorrectly, it is using `files` as `includes` was intended to be used.

I am likely going to release a patch release to Funnel, that warns and "fixes the problem".

@jasonmit do you think you can release a patch version of ember-intl that fixes this issue?

**Comment by stefanpenner** [comment]
> @jasonmit do you think you can release a patch version of ember-intl that fixes this issue?

actually PR in-bound

**Comment by stefanpenner** [comment]
and issue on funnel https://github.com/broccolijs/broccoli-funnel/issues/37

**Comment by demersus** [comment]
Thanks for your help on this

**Comment by stefanpenner** [comment]
ember-intl fix: https://github.com/yahoo/ember-intl/pull/194

funnel patch next

**Comment by stefanpenner** [comment]
funnel patch https://github.com/broccolijs/broccoli-funnel/pull/38

**Comment by stefanpenner** [comment]
looks like ember intl shipped a fix, and i will cut a patch funnel once i get a review on it to help other users who have falling into this trap.

**Comment by seandeburca** [comment]
Your fix worked for me. Thanks a lot!

**Comment by NeilAllen** [comment]
How do I get the fix?

**Comment by kellyselden** [comment]
@NeilAllen try `rm -rf node_modules && npm i`

**Comment by VincentMolinie** [comment]
>  @NeilAllen try rm -rf node_modules && npm i
I tried that but it doesn't work for me.
I'm using Node v5.8.0 and ember-cli v2.3.0

**Comment by Dan503** [comment]
I was following the [quick start guide](https://guides.emberjs.com/v2.5.0/getting-started/quick-start/) and ran into this issue when I tried to run the `ember serve` command.

The server started but when I go to `http://localhost:4200/` it only says `Build error` and an error message.

```````````
ENOENT: no such file or directory, scandir 'C:\Users\blah-blah-blah\ember-quickstart\node_modules\ember-cli-ic-ajax\node_modules'
`````````````

It's also saying this in the console:

````````````
Future versions of Ember CLI will not support v4.4.4. Please update to Node 0.12 or io.js.
version: 0.2.7
```````````

I've updated my npm and node to the latest versions and run `ember update` and I still can't get past either of these error messages. :(

**Comment by Dan503** [comment]
oh, I know what the issue was now.

When I did the initial install I used this `npm install -g ember-cli` instead of `npm install -g ember-cli@2.5`.

I thought `npm install -g ember-cli` would automatically install the latest version or at least ember update would update me to the latest version but I guess not. :/

I started over with a fresh install making sure to add the `@2.5` this time and now it's working. :)

(I know I only need to do that step once, not for every project)
